### PR TITLE
feat(kit): add `TuiPdfFallback` for `TuiPdfViewer`

### LINKED ISSUE #1866: 🚀 - `PdfViewer` display message when inline PDF is disabled
### What is the affected URL?

https://taiga-ui.dev/components/pdf-viewer

### Description

Just to press any button that must show PDF.
Expected: PdfViewver with pdf in it.
Actual: pdf-file download starts.
Firefox 101, no addons. Windows 11 21H2. Google Chrome - no issue.
[Screen Record](https://disk.yandex.ru/i/OWXRlshMG5SL2g)

### Which browsers have you used?

- [ ] Chrome
- [X] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #1866

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/172903333-3dc1ea34-f114-47cd-829d-06a227e70970.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/components/pdf-viewer/pdf-viewer.style.less [hunk 1]
```diff
@@ -47,3 +47,13 @@
     width: 100%;
     height: 100%;
 }
+
+.t-fallback-message {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    font: var(--tui-font-text-m);
+    transform: translate(-50%, -50%);
```
Comment (nsbarsukov) [inline projects/kit/components/pdf-viewer/pdf-viewer.style.less:56]: use mixin `.center-all()` from `@import 'taiga-ui-local';`

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        @Inject(TUI_IS_CHROMIUM) private readonly isChromium: boolean,
+        @Inject(TUI_IS_FIREFOX) private readonly isFirefox: boolean,
+        @Inject(TUI_IS_WEBKIT) private readonly isWebkit: boolean,
+        @Inject(ViewContainerRef) private readonly viewContainerRef: ViewContainerRef,
+    ) {
+        this.element.nativeElement.style.setProperty('opacity', '0');
+    }
+
+    ngAfterViewInit(): void {
+        race(timer(1_000), fromEvent(this.iframe, 'load'))
```
Comment (nsbarsukov) [inline projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts:35]: add comment why we need `timer(1000)`

File: projects/i18n/languages/russian/kit.ts [hunk 3]
```diff
@@ -53,4 +53,6 @@ export const TUI_RUSSIAN_LANGUAGE_KIT: LanguageKit = {
         drop: 'Поместите файл сюда',
         dropMultiple: 'Поместите файлы сюда',
     },
+    unavailablePdf:
+        'В настройках вашего браузера установлена загрузка файлов PDF вместо их отображения в окне',
```
Comment (nsbarsukov) [inline projects/i18n/languages/russian/kit.ts:57]: Let's ask our designer to come up with better description ? 

I'm not sure this is a clear explanation.

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/kit/components/pdf-viewer/pdf-viewer.style.less (hunk 1 above)
Comment (splincode) [reply projects/kit/components/pdf-viewer/pdf-viewer.style.less]: fixed

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts (hunk 2 above)
Comment (splincode) [reply projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts]: fixed

File: projects/i18n/languages/russian/kit.ts (hunk 3 above)
Comment (splincode) [reply projects/i18n/languages/russian/kit.ts]: fixed

**Comment by splincode** [comment]
@waterplea hello, please review me

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        @Inject(TUI_IS_FIREFOX) private readonly isFirefox: boolean,
+        @Inject(TUI_IS_WEBKIT) private readonly isWebkit: boolean,
+        @Inject(ViewContainerRef) private readonly viewContainerRef: ViewContainerRef,
+    ) {
+        this.element.nativeElement.style.setProperty('opacity', '0');
+    }
+
+    ngAfterViewInit(): void {
+        race(
+            timer(TIMEOUT), // @note: firefox doesn't trigger load event when pdf doesn't open
```
Comment (waterplea) [inline projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts:38]: What if it's a huge PDF that take a lot of time to load?

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                }
+            });
+    }
+
+    private get iframe(): HTMLIFrameElement {
+        return this.element.nativeElement;
+    }
+
+    private hasPdfInsideIframe(iframe: HTMLIFrameElement): boolean {
+        const contentWindow: Window = iframe.contentWindow || ({length: 0} as Window);
```
Comment (waterplea) [inline projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts:62]: This function doesn't look that trustworthy :D

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+             */
+            return contentWindow.length > 0;
+        } else if (this.isFirefox) {
+            /**
+             * @note: in firefox in the blank window hasn't length/then fields
+             */
+            return 'length' in contentWindow && 'then' in contentWindow;
+        } else if (this.isWebkit) {
+            /**
+             * @note: in safari/chrome ios in the blank window hasn't embedded element
```
Comment (waterplea) [inline projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts:77]: Does any mobile browser even allow embedding of PDF?

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts (hunk 4 above)
Comment (splincode) [reply projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts:38]: The bottom line is that the file size does not affect, I checked, the bottom line is to make sure that browser system elements are rendered inside the iframe and check it

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts (hunk 5 above)
Comment (splincode) [reply projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts:62]: But it's the only way I've determined :D

File: projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts (hunk 4 above)
Comment (waterplea) [reply projects/kit/directives/pdf-fallback/pdf-fallback.directive.ts:38]: Got it, I guess PDF viewer loads first and then loads its content.

**Pull request closed without merging**